src/PyDictProxyHandler.cc

```diff
@@ -168,13 +95,7 @@ bool PyDictProxyHandler::getOwnEnumerablePropertyKeys(
 }
 
 void PyDictProxyHandler::finalize(JS::GCContext *gcx, JSObject *proxy) const {
-  // We cannot call Py_DECREF here when shutting down as the thread state is gone.
-  // Then, when shutting down, there is only on reference left, and we don't need
-  // to free the object since the entire process memory is being released.
-  PyObject *self = JS::GetMaybePtrFromReservedSlot<PyObject>(proxy, PyObjectSlot);
-  if (Py_REFCNT(self) > 1) {
-    Py_DECREF(self);
-  }
+  return handleFinalize(proxy);
```

Comment:
just don't override the method at all and let it inherit the parent class version

Reply:
done